apps/web/lib/trpc/routers/newsletter.ts

```diff
@@ -0,0 +1,44 @@
+import { env } from "@/lib/env";
+import { TRPCError } from "@trpc/server";
+import { Resend } from "resend";
+import { z } from "zod";
+import { t } from "../trpc";
+
+export const newsletterRouter = t.router({
+  signup: t.procedure
+    .input(
+      z.object({
+        email: z.string(),
+      }),
+    )
+    .mutation(async ({ input }) => {
+      const { RESEND_API_KEY, RESEND_AUDIENCE_ID } = env();
+
+      if (!RESEND_API_KEY || !RESEND_AUDIENCE_ID) {
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "An unexpected error occurred, please try again later.",
+        });
+      }
+
+      const email = input.email;
+      if (!email) {
```

Comment:
we don't need this check, by using a zod schema in the input function above, it is already checked